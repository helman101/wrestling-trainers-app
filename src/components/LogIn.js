import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast, ToastContainer } from 'react-toastify';
import { userRequest } from '../API/api';
import styles from '../assets/styles/style.module.css';

const LogIn = (props) => {
  const [userEmail, setUserEmail] = useState('');
  const [userPassword, setUserPassword] = useState('');

  const handleChange = (e) => {
    const { name, value } = e.target;

    if (name === 'email') {
      setUserEmail(value);
    } else {
      setUserPassword(value);
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const params = { email: userEmail, password: userPassword };
    props.dispatch(userRequest(params, props.history.push, toast));
  };

  return (
    <div className={`${styles.login} ${styles.dFlex} ${styles.flexColumn} ${styles.justifyContentCenter}`}>
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />
      <div className={`${styles.sessionTitle}`}>Log In</div>
      <form className={`${styles.mt2} ${styles.form} ${styles.dFlex} ${styles.flexColumn} ${styles.justifyContentCenter}`}>
        <input type="email" id="email" name="email" placeholder="Email" onChange={handleChange} />
        <br />
        <input type="password" id="password" name="password" placeholder="Password" onChange={handleChange} />
        <br />
        <button type="submit" onClick={handleSubmit} className={`${styles.formBtn}`}>Submit</button>
      </form>
    </div>
  );
};

LogIn.propTypes = {
  dispatch: PropTypes.func.isRequired,
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
};

export default connect()(LogIn);
