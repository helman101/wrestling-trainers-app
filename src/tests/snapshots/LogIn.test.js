import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import defaultState from '../../store/store';
import LogIn from '../../components/LogIn';

const mockStore = configureStore([]);

describe('LogIn', () => {
  let store;
  let component;
  let history;

  beforeEach(() => {
    history = createMemoryHistory();
    store = mockStore(defaultState);
    component = renderer.create(
      <Provider store={store}>
        <LogIn history={history} />
      </Provider>,
    );
  });

  it('should render propertly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
