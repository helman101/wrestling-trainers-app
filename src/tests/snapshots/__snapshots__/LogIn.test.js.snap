// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should render propertly 1`] = `
<div
  className="login dFlex flexColumn justifyContentCenter"
>
  <div
    className="Toastify"
  />
  <div
    className="sessionTitle"
  >
    Log In
  </div>
  <form
    className="mt2 form dFlex flexColumn justifyContentCenter"
  >
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <br />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <br />
    <button
      className="formBtn"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
